src/middleware/statement.validation.middleware.ts

```diff
@@ -4,67 +4,77 @@ import { checkActiveBOExists, checkActiveMOExists, getApplicationData, hasAddedO
 import {
   FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION,
   UPDATE_CHECK_YOUR_ANSWERS_URL,
-  UPDATE_REVIEW_STATEMENT_URL
+  UPDATE_REVIEW_STATEMENT_URL,
+  SECURE_UPDATE_FILTER_URL,
 } from '../config';
 import { BeneficialOwnerStatementKey, BeneficialOwnersStatementType } from "../model/beneficial.owner.statement.model";
 import { ApplicationData } from "../model";
 import { ErrorMessages } from "../validation/error.messages";
 import { Session } from "@companieshouse/node-session-handler";
 import { RegistrableBeneficialOwnerKey } from "../model/update.type.model";
+import { yesNoResponse } from "../model/data.types.model";
 
-export const hasValidStatements = (req: Request, res: Response, next: NextFunction) => {
-  const errorList: string[] = [];
-  const appData: ApplicationData = getApplicationData(req.session as Session);
-  if (
-    isActiveFeature(FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION) &&
-    !checkStatementsValid(appData, errorList)
-  ) {
-    req['statementErrorList'] = errorList;
+export const statementValidationErrorsGuard = (req: Request, res: Response, next: NextFunction) => {
+  const hasStatementErrors = req['statementErrorList']?.length;
+  const flagEnabled = isActiveFeature(FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION);
+
+  if (flagEnabled && hasStatementErrors) {
     return next();
   }
 
+  const appData: ApplicationData = getApplicationData(req.session as Session);
   const redirectUrl = appData.update?.no_change ? UPDATE_REVIEW_STATEMENT_URL : UPDATE_CHECK_YOUR_ANSWERS_URL;
+
   return res.redirect(redirectUrl);
 };
 
-const validateIdentifiedBOsStatement = (appData: ApplicationData, errorList: string[]): boolean => {
-  const allOrSomeBOsIdentified: boolean = (appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.ALL_IDENTIFIED_ALL_DETAILS || appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.SOME_IDENTIFIED_ALL_DETAILS);
-  const someOrNoneBOsIdentified: boolean = (appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.SOME_IDENTIFIED_ALL_DETAILS || appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.NONE_IDENTIFIED);
-  const allBOsIdentified: boolean = (appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.ALL_IDENTIFIED_ALL_DETAILS);
-
-  if (allOrSomeBOsIdentified && !checkActiveBOExists(appData)) {
-    errorList.push(ErrorMessages.NO_ACTIVE_REGISTRABLE_BO);
+export const summaryPagesGuard = (req: Request, res: Response, next: NextFunction) => {
+  const hasStatementErrors = req['statementErrorList']?.length;
+  const flagEnabled = isActiveFeature(FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION);
+  if (!flagEnabled || flagEnabled && !hasStatementErrors) {
```

Comment:
you don't need second flagEnabled
 if (!flagEnabled || !hasStatementErrors) {

Reply:
Indeed 🤦‍♂️. Removed 👍